packages/peregrine/src/hooks/useDOMQuery.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const findDOMElements = queryString =>
+    document.querySelectorAll(queryString) || [];
+
+/**
+ * A custom hook that will return the array of elements that
+ * match the queryString. document.querySelectorAll is used
+ * to find all elements.
+ *
+ * @param {string} queryString
+ * @return Array<HTMLElement> and {
+ *  setInnerHTML: (string) => void,
+ *  setInnerText: (string) => void,
+ *  setAttribute: (string, any) => void
+ * }
+ */
+export const useDOMQuery = queryString => {
+    const [elements, setElements] = useState([]);
+    useEffect(() => {
+        setElements(findDOMElements(queryString));
+    }, []);
+    const setInnerHTML = useCallback(
```

Comment:
What if the caller wants to delete the contents of the element?

Reply:
> User can set it to the empty string '' to delete all content.

Careful, in JavaScript the empty string is falsy 😉

Reply:
@zetlen is asking the right question here. Re-implementing a subset of the `Element` API as callbacks is a mistake; this is an incomplete abstraction, and to complete it we'd have to add more and more callbacks until we covered 100% of the API, at which point we would have a large amount of code adding no value.

Reply:
My bad. Fixed it. But you can still send an empty string to clear.